lib/web_ui/dev/test_runner.dart

```diff
@@ -240,12 +245,40 @@ class TestCommand extends Command<bool> with ArgUtils {
     }
 
     if (htmlTargets.isNotEmpty) {
-      await _buildTests(targets: htmlTargets, forCanvasKit: false);
+      await _buildTestsInParallel(targets: htmlTargets, forCanvasKit: false);
     }
 
+    // Currently iOS Safari tests are running on simulator, which does not
+    // support canvaskit backend.
     if (canvasKitTargets.isNotEmpty) {
-      await _buildTests(targets: canvasKitTargets, forCanvasKit: true);
+      await _buildTestsInParallel(
+          targets: canvasKitTargets, forCanvasKit: true);
+    }
+
+    // Copy image files under build directory.
+    // A side effect is this file copies all the images btween, even only one target
+    // test is asked to run.
+    final List<io.FileSystemEntity> contents =
+        environment.webUiTestDir.listSync(recursive: true);
+    for (final io.FileSystemEntity entity in contents) {
```

Comment:
tip: you can do `contents.whereType<io.File>()`, then you won't need the `if` block inside.

Reply:
done. thanks!